web/gds-admin-ui/src/pages/app/details/BasicDetails/components/BasicDetailsDropdown.js

```diff
@@ -0,0 +1,94 @@
+    const reviewNotes = useSelector(getAllReviewNotes)
+
+    const handleClose = () => dispatch({ type: actionType.SEND_EMAIL_MODAL, payload: { vasp: { name: vasp?.name, id: vasp?.vasp?.id } } })
+
+    const generatePdfDocument = async (filename) => {
+        nProgress.start()
+        try {
+            const blob = await pdf(<VaspDocument vasp={vasp} notes={reviewNotes} />).toBlob()
+            downloadFile(blob, `${filename}.pdf`, 'application/pdf')
+            nProgress.done()
+        } catch (error) {
+            console.error('Unable to export as PDF', error)
+            nProgress.done()
+        }
+    };
+
+    return (
+        <Dropdown className="float-end" align="end">
+            <Dropdown.Toggle
+                data-testid="dripicons-dots-3"
+                variant="link"
+                tag="a"
+                className="card-drop arrow-none cursor-pointer p-0 shadow-none">
+                <i className="dripicons-dots-3"></i>
+            </Dropdown.Toggle>
+            <Dropdown.Menu>
+                <Dropdown.Item data-testid="reviewItem" disabled={isNotPendingReview()}>
+                    <i className="mdi mdi-card-search me-1"></i>Review
+                </Dropdown.Item>
+
+                {
+                    isEditMenuAvailable(vasp?.vasp?.verification_status) && (
+                        <>
+                            <Modal>
+                                <ModalOpenButton>
+                                    <Dropdown.Item>
+                                        <i className="mdi mdi-briefcase-edit me-1"></i>Edit Business Info
+                                    </Dropdown.Item>
+                                </ModalOpenButton>
+                                <ModalContent size="lg">
+                                    <Row className='p-4'>
+                                        <BusinessInfosForm data={vasp} />
+                                    </Row>
+                                </ModalContent>
+                            </Modal >
+                            <Modal>
+                                <ModalOpenButton>
+                                    <Dropdown.Item>
+                                        <i className="mdi mdi-network me-1"></i>Edit TRISA Details
+                                    </Dropdown.Item>
+                                </ModalOpenButton>
+                                <ModalContent size="lg">
+                                    <Row className='p-4'>
+                                        <TrisaImplementationDetailsForm data={vasp} />
+                                    </Row>
+                                </ModalContent>
+                            </Modal >
+                            <Dropdown.Item>
+                                <i className="mdi mdi-office-building me-1"></i>Edit IVMS 101 Record
+                            </Dropdown.Item>
+                        </>
+                    )
+                }
```

Comment:
This should only cover the `Edit TRISA Details` menu item, the IVMS 101 record and Business info can be edited at any time.

Reply:
alright